### PR TITLE
simplify the registry interface used by the farm

### PR DESCRIPTION
**Pull request by mpuncel**

### CODE REVIEW

File: pkg/rc/farm.go [hunk 1]
```diff
@@ -89,7 +88,7 @@ func NewFarm(
 // Set the metrics registry to use (see
 // https://godoc.org/github.com/rcrowley/go-metrics#Registry). If set,
 // performance metrics will be recorded to the registry.
-func (rcf *Farm) SetMetricsRegistry(registry metrics.Registry) error {
+func (rcf *Farm) SetMetricsRegistry(registry rcmetrics.MetricsRegistry) error {
```
Comment (anthonybishopric) [inline pkg/rc/farm.go]: Out of curiosity, why isn't this method `SetMetrics(metrics rcmetrics.Metrics)`?

Comment (mpuncel) [inline pkg/rc/farm.go]: what's the benefit of that? That would also work but i guess i like the logging in the implementation of `RecordRCProcessingTime`, and we would probably mutate the `rcmetrics.Metrics` instance passed in to layer on metrics

Comment (mpuncel) [inline pkg/rc/farm.go]: Is it that you think people might want to do things with rc_processing_time other than put them in a Histogram?

Comment (anthonybishopric) [inline pkg/rc/farm.go]: Maybe this is a broader question - why does anything in this file need to be aware of the underlying `rcmetrics.MetricsRegistry`? As far as I can tell, it's an implementation detail of `Metrics`, not of the RC farm. The logger appears to be passed in directly without alteration from the init of the farm to the init of `&rcmetrics.Metrics{}` on line 82.

Comment (mpuncel) [inline pkg/rc/farm.go]: that actually might be easier for my use case though now that I think about it.

**Comment by anthonybishopric** [comment]
:+1:

**Pull request merged**